### ISSUE TITLE
Could not update the provider: The discovery endpoint is not reachable.

### ISSUE THREAD

**Issue by Snuffy2** [nextcloud/user_oidc#803]
I'm not sure where the issue is but I'm unable to get this working. I'm using Authentik for OIDC.

The Configuration URL appears to be correct since when I open it in a browser it is loading this (domain name replaced): 
[openid-configuration.json](https://github.com/nextcloud/user_oidc/files/14455666/openid-configuration.json)

I've triple checked the Config Id and Secret are correct and even removed and regenerated new ones in Authentik to be sure.

Happy to provide any other logs or try anything else suggested.

**Comment by tugdualenligne** [comment]
Same issue here. Authentik version 2024.2.2 and Nextcloud 28.0.2

**Comment by abjoseph** [comment]
Same issue here as well. Authentik version 2024.2.2 and Nextcloud 28.0.3

**Comment by julien-nc** [comment]
You can check if you see `Discovery endpoint validation error` in the nextcloud server logs. These lines should contain more information about why the discovery endpoint can't be reached.

If there is no such line, it most likely means the network request from NC to the discovery endpoint failed. I could give you a patch to get more information about that in the logs.

**Comment by abjoseph** [comment]
@julien-nc I found this [comment ](https://github.com/nextcloud/user_oidc/issues/555#issuecomment-1379674153) and applying the same setting, it resolved the error reported by OP. Not saying it's the resolution for OP but it worked for me. Maybe something that should be documented in the main README.

**Comment by tugdualenligne** [comment]
Thanks
I’ve got this setting already in my config.php file and it doesn’t resolve
my issue


Le ven. 29 mars 2024 à 22:52, Brandon Joseph ***@***.***> a
écrit :

**Comment by 4sens** [comment]
Having the same issue here. Any updates?

**Comment by edward-ly** [comment]
Hi, if anyone can report whether the issue still persists on the latest versions of Authentik/Nextcloud/user_oidc and provide some reproducible steps if so, that would be greatly appreciated. We'll take a look if the issue is confirmed to be on our end.

**Comment by Paco93** [comment]
Having the same issue here. I am on latest versions of Authentik/Nextcloud/user_oidc. 
Logging says:

Error | user_oidc | LocalServerExceptionHost "192.168.1.230" (xx.xx.xx.xx:80) violates local access rulesDiscovery endpoint validation error


I added in config.php 'allow_local_remote_servers' => 'true' but no  success.

**Comment by edward-ly** [comment]
If you're running Nextcloud/Authentik inside Docker, it's possible that local IP addresses aren't being recognized correctly from within the container. Try running `ip a` and using one of the IP addresses listed that is assigned to Docker instead for all containers.

**Comment by Paco93** [comment]
Yes I am running nextcloud/authentik under docker but on different machines (actually different lxc containers) and with a proxy manager (NPM) on another machine on my lan.

**Comment by niallreenan** [comment]
OK, so I will give a possible solution to this issue and the backchannel logout Keycloak issue.

I have been driving myself crazy for the last week and spent about 50 hrs trying to get a Keycloak server talking to a Nextcloud server. If you just want OIDC SSO then I would recommend the other Nextcloud plugin based on the Social Login plugin. It can be set up very easily and will handle logins without much hassle.

However, if you need backchannel logout as we did, then this plugin is the only easy solution unless you want to code your own custom backchannel logout endpoint (also, thank you so much for this plugin!!).

In order to get rid of the "The discovery endpoint is not reachable" you will need to make sure both the Keycloak server **and** the Nextcloud server have SSL certs and HTTPS enabled. In order to make this work correctly you will **also** need the Keycloak server running with a production database (see Keycloak documentation). The trick part is that to get this plugin working for us, **Keycloak and Nextcloud both needed to have the other's SSL cert added to their trusted SSL certs**. It won't work if Nextcloud only has the Keycloak cert, both servers will need to have the other servers' certs.

After I did this the "The discovery endpoint is not reachable" error went away. Then with some extra configuring, you can add the Nextcloud backchannel logout URL to Keycloak. The trick here was making sure "front channel logout" is unticked. There is a bug in the most recent version of Keycloak that blocks the backchannel logout if that option is turned on.

Now Keycloak and Nextcloud will correctly handle both logins and logouts from any client.

Hope this helps someone avoid the hell I've had this week :(

**Comment by seakayone** [comment]
I have had the same issue.

Using nextcloud 32.0.2 with the OIDC user backend 8.1.0 and authentik 2025.10.2.

My setup is so that all services run as containers using docker compose. I have a https domain with letsencrypt certificates using traefik to route subdomains. Since my ISP does not provide public IPv4 addresses the domain is only reachable with IPV6.

If I call the OIDC discovery url in the browser it resolves just fine and returns correctly. 

When registering the OIDC provider in the backend in nextcloud I receive the error message "The discovery endpoint is not reachable".

The logs of nextcloud do not show the cause of the error:
```
nextcloud-app    | 172.19.0.1 - - [06/Dec/2025:11:21:43 +0000] "POST /login/confirm HTTP/1.1" 200 841 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:145.0) Gecko/20100101 Firefox/145.0"
nextcloud-app    | 172.19.0.1 - - [06/Dec/2025:11:21:43 +0000] "POST /apps/user_oidc/provider HTTP/1.1" 400 756 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:145.0) Gecko/20100101 Firefox/145.0"
nextcloud-app    | 172.19.0.1 - - [06/Dec/2025:11:21:47 +0000] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.1" 304 692 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:145.0) Gecko/20100101 Firefox/145.0"
```

Executing in the nextcloud container resolved the auth domain's AAAA record to the correct IPv6 address.

I was able to use authentik with other services with this exact setup.

Adding `extra_hosts` entries in the docker compose did not solve my problem alone.

I have been able to register the provider only after adding the local IPv4 address of my docker host to the docker compose 
```
   extra_hosts:
     - "auth.example.com:192.168.1.1"
```
and adding the `'allow_local_remote_servers' => 'true'`  to config.php

**Comment by rainerwein** [comment]
I have the same issue with Nextcloud 32.0.5 and Pocket ID running in seperate docker compose stacks. 
From the nextcloud container shell I can curl the discovery location via https (with letsencrypt certs) with no errors.
I have this error in my nextcloud logs
`172.XX.XX.XX - - [12/Feb/2026:10:21:16 +0000] "POST /apps/user_oidc/provider HTTP/1.1" 400 757 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:147.0) Gecko/20100101 Firefox/147.0"`

**Comment by murmelbahn** [comment]
> I have had the same issue.
> 
> Using nextcloud 32.0.2 with the OIDC user backend 8.1.0 and authentik 2025.10.2.
> 
> My setup is so that all services run as containers using docker compose. I have a https domain with letsencrypt certificates using traefik to route subdomains. Since my ISP does not provide public IPv4 addresses the domain is only reachable with IPV6.
> 
> If I call the OIDC discovery url in the browser it resolves just fine and returns correctly.
> 
> When registering the OIDC provider in the backend in nextcloud I receive the error message "The discovery endpoint is not reachable".
> 
> The logs of nextcloud do not show the cause of the error:
> 
> ```
> nextcloud-app    | 172.19.0.1 - - [06/Dec/2025:11:21:43 +0000] "POST /login/confirm HTTP/1.1" 200 841 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:145.0) Gecko/20100101 Firefox/145.0"
> nextcloud-app    | 172.19.0.1 - - [06/Dec/2025:11:21:43 +0000] "POST /apps/user_oidc/provider HTTP/1.1" 400 756 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:145.0) Gecko/20100101 Firefox/145.0"
> nextcloud-app    | 172.19.0.1 - - [06/Dec/2025:11:21:47 +0000] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.1" 304 692 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:145.0) Gecko/20100101 Firefox/145.0"
> ```
> 
> Executing in the nextcloud container resolved the auth domain's AAAA record to the correct IPv6 address.
> 
> I was able to use authentik with other services with this exact setup.
> 
> Adding `extra_hosts` entries in the docker compose did not solve my problem alone.
> 
> I have been able to register the provider only after adding the local IPv4 address of my docker host to the docker compose
> 
> ```
>    extra_hosts:
>      - "auth.example.com:192.168.1.1"
> ```
> 
> and adding the `'allow_local_remote_servers' => 'true'` to config.php

Thanks! This works for me. I'm using Pocket ID and Nextcloud from Linuxserver.io in Docker.

**Comment by kmoninsis** [comment]
I ran into this issue, with Nextcloud behind Traefik reverse proxy, and made it work by adding the IP of the Traefik container instead of the docker host IP to the `extra_hosts`
